tiledb/sm/serialization/array.cc

```diff
@@ -58,6 +58,13 @@ namespace tiledb {
 namespace sm {
 namespace serialization {
 
+class ArraySerializationStatusException : public StatusException {
+ public:
+  explicit ArraySerializationStatusException(const std::string& message)
```

Comment:
`ArraySerializationException` is sufficient. That it comes from a `StatusException` is immaterial.